src/mem/old_api.cc

```diff
@@ -456,15 +457,29 @@ Mem::Init(void)
     Mgr::RegisterAction("mem", "Memory Utilization", Mem::Stats, 0, 1);
 }
 
-void
-Mem::Report()
+namespace Mem
+{
+
+static void
+LogPoolSummary()
 {
     debugs(13, 3, "Memory pools are '" <<
            (Config.onoff.mem_pools ? "on" : "off")  << "'; limit: " <<
            std::setprecision(3) << toMB(MemPools::GetInstance().idleLimit()) <<
            " MB");
 }
 
```

Comment:
Please completely remove Mem::Report() instead. 

The official code should not call Mem::Report(). There is no justification for main.cc treating memory module in such a special way. The PR code abuses finalizeConfig() and syncConfig() APIs for reporting configuration details. The reporting code itself has problems. Removing all this code is better than leaving or polishing it.

These level-3 details are meant for developers. Developers can use the existing configuration debugging to extract very similar configuration details. If somebody actually needs more debugging, let's add it to MemPools::setIdleLimit() and/or other configuration methods, where it belongs.

Reply:
FTR; a lot of non-developers use levels 3-9 for behaviour analysis, flow monitoring and other use-cases.

Anyhow, testing complete removal and the associated cleanup that allows now. Will re-focus this PR when that passes.

Reply:
Done as of [d8b4186](https://github.com/squid-cache/squid/pull/1184/commits/d8b4186f3fcd8f033ea9ceadbf5fcd6115f9e7cf)

Reply:
> FTR; a lot of non-developers use levels 3-9 for behaviour analysis, flow monitoring and other use-cases.

Just like developers, those folks can use the existing configuration debugging to extract very similar configuration details. If somebody actually needs more debugging, let's add it to MemPools::setIdleLimit() and/or other configuration methods, where it belongs.

Said that, I think we should not go the extra mile to accommodate "non-developer" needs in level-3+ debugging logs. To improve debugging interface and its performance, we should focus on satisfying "developer" needs instead. Otherwise, nobody will be happy, neither "developers" nor "non-developers" that read level-3+ debugging logs.